### PR TITLE
feat(inputs.rabbitmq): Add secretstore support for username and password

### PR DESCRIPTION
**Pull request by yonathanamir**
- [x] Updated associated README.md.
- [ ] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

resolves https://github.com/influxdata/telegraf/issues/13956

Adds secretstore support to the username and password options of the rabbitmq input.

### CODE REVIEW

File: plugins/inputs/rabbitmq/rabbitmq.go [hunk 1]
```diff
@@ -350,17 +350,29 @@ func (r *RabbitMQ) requestEndpoint(u string) ([]byte, error) {
...
+	username, err := r.Username.Get()
+	if err != nil {
+		return nil, err
+	}
+	defer username.Destroy()
+	usernameStr := username.String()
+
+	if usernameStr == "" {
+		usernameStr = DefaultUsername
+	}
```
Comment (srebhan) [inline plugins/inputs/rabbitmq/rabbitmq.go:362]: You can directly ask the secret if it is empty...
```suggestion
	username := DefaultUsername
	if !r.Username.Empty() {
		usr, err := r.Username.Get()
		if err != nil {
			return nil, err
		}
		defer usr.Destroy()
		username = usr.String()
	}
```
What do you think? If you like it, the same can be done for password.

**Review by srebhan — COMMENTED** [review]
Thanks for splitting the PR @yonathanamir! One small suggestion...

File: plugins/inputs/rabbitmq/rabbitmq.go (hunk 1 above)
Comment (yonathanamir) [reply plugins/inputs/rabbitmq/rabbitmq.go]: Looks good, I've updated the PR.

**Review by srebhan — APPROVED** [review]
Thanks for converting the plugin @yonathanamir! Much appreciated!

**Review by powersj — APPROVED** [review]

**Pull request merged**